dimod/generators/gates.py

```diff
@@ -13,10 +13,12 @@
 #    limitations under the License.
 
 from collections import defaultdict
-
+from typing import Optional
+import itertools as it
```

Comment:
To be consistent with the rest of dimod, I think it's better to use `import itertools` or `from itertools import product, starmap`.

Also while we're at it we should follow pep8's [import ordering](https://www.python.org/dev/peps/pep-0008/#imports) specifically
1) Standard library imports.
2) Related third party imports.
3) Local application/library specific imports.